biom/table.py

```diff
@@ -948,6 +948,65 @@ def _get_col(self, col_idx):
         self._data = self._data.tocsc()
         return self._data.getcol(col_idx)
 
+    def align_metadata(self, metadata, axis='sample'):
+        """ Aligns metadata against biom table, only keeping common ids.
+
+        Parameters
+        ----------
+        metadata : pd.DataFrame
+            The metadata, either respect to the sample metadata
+            or observation metadata.
+        axis : {'sample', 'observation'}
+            The axis on which to operate.
+
+        Returns
+        -------
+        biom.Table
+            A filtered biom table.
+        pd.DataFrame
+            A filtered metadata table.
+        """
+        ids = set(self.ids(axis)) & set(metadata.index)
+        filter_f = lambda v, i, m: i in ids
+        t = self.table.filter(filter_f, axis=axis, inplace=False)
+        md = metadata.loc[self.ids()]
```

Comment:
Should this be `t.ids()` as the filter is not performed inplace?

Reply:
correct.  It actually should be `t.ids(axis=axis)`